firefox-ios/Client/TabManagement/Legacy/LegacyTabManager.swift

```diff
@@ -819,56 +824,96 @@ class LegacyTabManager: NSObject, FeatureFlaggable, TabManager, TabEventHandler
         privateConfiguration.preferences.javaScriptCanOpenWindowsAutomatically = allowPopups
     }
 
-    // returns all activate tabs (private or normal)
-    private func viableTabs(isPrivate: Bool = false) -> [Tab] {
-        if !isPrivate, featureFlags.isFeatureEnabled(.inactiveTabs, checking: .buildAndUser) {
-            // only use active tabs as viable tabs
-            // we cannot use recentlyAccessedNormalTabs as this is filtering for sponsored and sorting tabs
-            return LegacyInactiveTabViewModel.getActiveEligibleTabsFrom(normalTabs)
-        } else {
-            return isPrivate ? privateTabs : normalTabs
+    private func updateSelectedTabAfterRemovalOf(_ removedTab: Tab, deletedIndex: Int) {
+        // We have to handle 3 different types of tabs: private tabs, normal active tabs, and normal inactive tabs.
+        // These tabs are all in the `tabs` array but are differentiated by their respective flags.
+        //
+        // Once a tab has been removed, we must consider several situations:
+        // A) A change in the `tabs` array size, which may require us to shift the selected tab index 1 to the left
+        // B) If the removed tab was selected, we must choose an appropriate next selected tab from among the existing tabs
+        // C) When we close the last tab of a certain type, we may need to perform additional logic
+        //      - For example, closing the last private tab should select the most recent active normal tab, if possible
+
+        // Viable tabs for selection are private tabs and normal active tabs. We never want to select a normal inactive tab.
+        let viableTabsToSelect = removedTab.isPrivate
+                          ? privateTabs
+                          : normalActiveTabs
```

Comment:
I really like this spacing for ternary operator syntax btw, much easier to parse for me